packages/pieces/community/pinterest/src/lib/triggers/new-pin-on-board.ts

```diff
@@ -0,0 +1,64 @@
+
+import { createTrigger, TriggerStrategy, PiecePropValueSchema, Property } from '@activepieces/pieces-framework';
+import { DedupeStrategy, Polling, pollingHelper } from '@activepieces/pieces-common';
+import dayjs from 'dayjs';
+import { makeRequest } from '../common';
+import { pinterestAuth } from '../common/auth';
+import { HttpMethod } from '@activepieces/pieces-common';
+
+const polling: Polling<PiecePropValueSchema<typeof pinterestAuth>, Record<string, any>> = {
+    strategy: DedupeStrategy.TIMEBASED,
+    items: async ({ propsValue, auth, lastFetchEpochMS }) => {
+        const board_id = propsValue['board_id'];
```

Comment:
he board_id should use the dynamic dropdown instead of a text input:
```suggestion
           board_id: boardIdDropdown
```